typescript/cli/src/deploy/warp.ts

```diff
@@ -381,3 +384,131 @@ export async function runWarpRouteApply(params: ApplyParams) {
     }),
   );
 }
+
+function displayWarpDeployPlan(deployConfig: WarpRouteDeployConfig) {
+  logBlue('\nWarp Route Deployment Plan');
+  logGray('==========================');
+  log(`📋 Token Standard: ${deployConfig.isNft ? 'ERC721' : 'ERC20'}`);
+
+  const { transformedDeployConfig, transformedIsmConfigs } =
+    transformDeployConfigForDisplay(deployConfig);
+
+  log('📋 Warp Route Config:');
+  logTable(transformedDeployConfig);
+  objMap(transformedIsmConfigs, (chain, ismConfigs) => {
+    log(`📋 ${chain} ISM Config(s):`);
+    ismConfigs.forEach((ismConfig) => {
+      logTable(ismConfig);
+    });
+  });
+}
+
+/* only used for transformIsmForDisplay type-sense */
+type IsmConfig =
+  | RoutingIsmConfig // type, owner, ownerOverrides, domain
+  | AggregationIsmConfig // type, modules, threshold
+  | MultisigIsmConfig // type, validators, threshold
+  | OpStackIsmConfig // type, origin, nativeBridge
+  | PausableIsmConfig // type, owner, paused, ownerOverrides
+  | TrustedRelayerIsmConfig; // type, relayer
+
+function transformDeployConfigForDisplay(deployConfig: WarpRouteDeployConfig) {
+  const transformedIsmConfigs: Record<ChainName, any[]> = {};
+  const transformedDeployConfig = objMap(deployConfig, (chain, config) => {
+    if (config.interchainSecurityModule)
+      transformedIsmConfigs[chain] = transformIsmConfigForDisplay(
+        config.interchainSecurityModule as IsmConfig,
+      );
+
+    return {
+      'NFT?': config.isNft ?? false,
+      Type: config.type,
+      Owner: config.owner,
+      Mailbox: config.mailbox,
+      'ISM Config(s)': config.interchainSecurityModule
+        ? 'See table(s) below.'
+        : 'No ISM config(s) specified.',
+    };
+  });
+
+  return {
+    transformedDeployConfig,
+    transformedIsmConfigs,
+  };
+}
+
+function transformIsmConfigForDisplay(ismConfig: IsmConfig): any[] {
+  const ismConfigs: any[] = [];
+  switch (ismConfig.type) {
+    case IsmType.AGGREGATION:
+      ismConfigs.push({
+        Type: ismConfig.type,
+        Threshold: ismConfig.threshold,
+        Modules: 'See table(s) below.',
+      });
+      ismConfig.modules.forEach((module) => {
+        ismConfigs.push(...transformIsmConfigForDisplay(module as IsmConfig));
+      });
+      return ismConfigs;
+    case IsmType.ROUTING:
+      return [
+        {
+          Type: ismConfig.type,
+          Owner: ismConfig.owner,
+          'Owner Overrides': ismConfig.ownerOverrides ?? 'Undefined',
+          Domains: ismConfig.domains,
```

Comment:
each `domain` is also an ismConfig, should we do something similar as the aggregation?

Reply:
decided to omit for brevity/speed since this display altogether is just a nice to have for the user